src/api/grainConfig.js

```diff
@@ -30,56 +28,24 @@ export const parser: C.Parser<GrainConfig> = C.object(
  * fields can form valid policies.
  */
 export function toDistributionPolicy(x: GrainConfig): DistributionPolicy {
-  const immediatePerWeek = NullUtil.orElse(x.immediatePerWeek, 0);
-  const recentPerWeek = NullUtil.orElse(x.recentPerWeek, 0);
-  const balancedPerWeek = NullUtil.orElse(x.balancedPerWeek, 0);
-
-  if (!isNonnegativeInteger(immediatePerWeek)) {
-    throw new Error(
-      `immediate budget must be nonnegative integer, got ${immediatePerWeek}`
-    );
-  }
-  if (!isNonnegativeInteger(recentPerWeek)) {
-    throw new Error(
-      `recent budget must be nonnegative integer, got ${recentPerWeek}`
-    );
+  if (!x.allocationPolicies.length) {
+    throw new Error(`Please provide at least one allocation policy, got zero.`);
   }
-  if (!isNonnegativeInteger(balancedPerWeek)) {
-    throw new Error(
-      `balanced budget must be nonnegative integer, got ${balancedPerWeek}`
-    );
-  }
-  const allocationPolicies = [];
-  if (immediatePerWeek > 0) {
-    allocationPolicies.push({
-      budget: G.fromInteger(immediatePerWeek),
-      policyType: "IMMEDIATE",
-    });
-  }
-  if (recentPerWeek > 0) {
-    const {recentWeeklyDecayRate} = x;
-    if (recentWeeklyDecayRate == null) {
-      throw new Error(`no recentWeeklyDecayRate specified for recent policy`);
+
+  x.allocationPolicies.map((policy) => {
+    if (!G.gt(policy.budget, G.ZERO)) {
+      throw new Error(
+        `${policy.policyType} budget must be positive integer, got ${policy.budget}`
+      );
```

Comment:
As above: a zero budget is totally fine.

Reply:
Done.